cloudformation/functions/get_group_role_map.py

```diff
@@ -0,0 +1,83 @@
+import boto3
+
+
+def get_paginated_results(product, action, key, credentials=None, args=None):
+    """Paginate through AWS API responses, combining them into a list
+
+    :param str product: AWS product name
+    :param str action: AWS API action name
+    :param str key: The parent key in the paginated response
+    :param dict credentials: Optional AWS API credentials
+    :param dict args: Optional additional arguments to pass to action method
+    :return: list of responses from all pages
+    """
+
+    args = {} if args is None else args
+    credentials = {} if credentials is None else credentials
+    return [y for sublist in [x[key] for x in boto3.client(
+        product, **credentials).get_paginator(action).paginate(**args)]
+            for y in sublist]
```

Comment:
I am always in favor of clever code, but I'm not sure `x`, `y`, `sublist`, keyword expansion, and double list comprehensions makes for particularly maintainable code.

Reply:
I expanded this and improved the variable names